org.emoflon.ibex.tgg.core.runtime/src/org/emoflon/ibex/tgg/compiler/BlackPatternCompiler.java

```diff
@@ -60,47 +67,89 @@ public void preparePatterns() {
 				factory.createBlackPattern(FWDFusedPattern.class);
 			}
 			factory.createBlackPattern(ConsistencyPattern.class);
-			
+
 			// ILP supported transformation
 			factory.createBlackPattern(FWDOptBlackPattern.class);
 			factory.createBlackPattern(BWDOptBlackPattern.class);
 			if (rule instanceof TGGComplementRule) {
 				factory.createBlackPattern(BWDOptFusedPattern.class);
 				factory.createBlackPattern(FWDOptFusedPattern.class);
 			}
-			
+
+			// Create edge patterns for all patterns found so far
+			for (IBlackPattern pattern : factory.getPatterns()) {
+				int minNrofEdges;

```

Comment:
Was macht solchen Code im Compiler?  Ich dachte die ganze Logik wäre doch nur im Optimiser.

Reply:
Der Optimizer kennt leider die IBex-Options nicht, außerdem gibt es den Optimizer nur für IbexBasePatterns. Deshalb kann ich den Code, der hier steht, leider nicht mit in den Optimizer packen